leasing/report/invoice/invoicing_review.py

```diff
@@ -0,0 +1,445 @@
+import datetime
+from collections import defaultdict
+from fractions import Fraction
+from operator import itemgetter
+
+from django.db import connection
+from django.utils.translation import pgettext_lazy
+from django.utils.translation import ugettext_lazy as _
+from enumfields import Enum
+from enumfields.drf import EnumField
+from rest_framework.response import Response
+
+from leasing.report.excel import ExcelCell, ExcelRow, FormatType
+from leasing.report.report_base import ReportBase
+
+
+class InvoicingReviewSection(Enum):
+    INVOICING_NOT_ENABLED = "invoicing_not_enabled"
+    RENT_INFO_NOT_COMPLETE = "rent_info_not_complete"
+    NO_RENTS = "no_rents"
+    NO_DUE_DATE = "no_due_date"
+    ONE_TIME_RENTS_WITH_NO_INVOICE = "one_time_rents_with_no_invoice"
+    INCORRECT_MANAGEMENT_SHARES = "incorrect_management_shares"
+    INCORRECT_RENT_SHARES = "incorrect_rent_shares"
+    NO_TENANT_CONTACT = "no_tenant_contact"
+    NO_LEASE_AREA = "no_lease_area"
+    INDEX_TYPE_MISSING = "index_type_missing"
+
+    class Labels:
+        INVOICING_NOT_ENABLED = pgettext_lazy(
+            "Invoicing review", "Invoicing not enabled"
+        )
+        RENT_INFO_NOT_COMPLETE = pgettext_lazy(
+            "Invoicing review", "Rent info not complete"
+        )
+        NO_RENTS = pgettext_lazy("Invoicing review", "No rents")
+        NO_DUE_DATE = pgettext_lazy("Invoicing review", "No due date")
+        ONE_TIME_RENTS_WITH_NO_INVOICE = pgettext_lazy(
+            "Invoicing review", "One time rents with no invoice"
+        )
+        INCORRECT_MANAGEMENT_SHARES = pgettext_lazy(
+            "Invoicing review", "Incorrect management shares"
+        )
+        INCORRECT_RENT_SHARES = pgettext_lazy(
+            "Invoicing review", "Incorrect rent shares"
+        )
+        NO_TENANT_CONTACT = pgettext_lazy("Invoicing review", "No tenant contact")
+        NO_LEASE_AREA = pgettext_lazy("Invoicing review", "No lease area")
+        INDEX_TYPE_MISSING = pgettext_lazy("Invoicing review", "Index type missing")
+
+
+INVOICING_REVIEW_QUERIES = {
```

Comment:
It could be nice that there would be some sort of documentation about the queries in Confluence. It will be kind of horror if someone else has to do some modification to this.